modules/dnn/src/layers/harmax.cpp

```diff
@@ -78,7 +75,7 @@ class LayerHardmaxImpl CV_FINAL : public HardmaxLayer
         MatShape shape(src.size.p, src.size.p + src.dims);
 
         // Prepare output
-        dst = Mat::zeros(src.dims, src.size, src.type());
+        memset(dst.ptr(), 0, dst.total() * dst.elemSize());
```

Comment:
The change looks redundant. `dst` may be mot contiguous.

Reply:
> dst may be mot contiguous.

Theoretically it is but I assume it is quite rare in dnn use cases. But anyway, zero can be set inside the loop.

Reply:
lets use ::zeros or at least add contigous check.